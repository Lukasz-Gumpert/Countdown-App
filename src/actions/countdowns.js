import uuid from 'uuid';
import database from '../firebase/firebase';

export const addCountdown = (countdown) => ({
  type: 'ADD_COUNTDOWN',
  countdown
});

export const startAddCountdown = (countdown = {}) => {
  return (dispatch) => {
    const {
      title = '',
      category = '',
      releaseDate = 0,
      description = '',
    } = countdown;
    const countdownData = { title, category, releaseDate, description };
    return database.ref('countdowns').push(countdownData).then((ref) => {
      dispatch(addCountdown({
        id: ref.key,
        ...countdownData
      }));
    });
  };
};

export const removeCountdown = ({ id }) => ({
  type: 'REMOVE_COUNTDOWN',
  id
});

export const editCountdown = (id, updates) => ({
  type: 'EDIT_COUNTDOWN',
  id,
  updates
});

export const setCountdowns = (countdowns) => ({
  type: 'SET_COUNTDOWNS',
  countdown
})