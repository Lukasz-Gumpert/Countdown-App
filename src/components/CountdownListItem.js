import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment'
import CountdownClock from './CountdownClock';

const CountdownListItem = ({ id, title, category, releaseDate, description }) => (
    <Link to={`edit/${id}`} className="list-item">
        <p> {category} </p>
        <h1>{title}</h1> : {description} <p>Time left:</p>
        <CountdownClock timeTillDate={moment(releaseDate)} />
    </Link>
);

export default CountdownListItem;