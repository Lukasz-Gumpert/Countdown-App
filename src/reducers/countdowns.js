const countdownsReducerDefaultState = [];

export default (state = countdownsReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_COUNTDOWN':
            return [
                ...state,
                action.countdown
            ];
        case 'EDIT_COUNTDOWN':
            return state.map((countdown) => {
                if (countdown.id === action.id) {
                    return {
                        ...countdown,
                        ...action.updates
                    }
                } else {
                    return countdown;
                };
            });
        case 'REMOVE_COUNTDOWN':
            return state.filter(({ id }) => {
                return id !== action.id
            });
        case 'SET_COUNTDOWNS':
            return action.countdowns;
        default:
            return state;
    };
};