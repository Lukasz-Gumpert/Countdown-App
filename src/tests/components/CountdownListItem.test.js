import React from 'react';
import { shallow } from 'enzyme';
import CountdownListItem from '../../components/CountdownListItem.js'
import countdowns from '../fixtures/countdowns';

test('Should render countdownListItem correctly', () => {
    const wrapper = shallow(<CountdownListItem countdowns={countdowns[0]} />);
    expect(wrapper).toMatchSnapshot();
});



