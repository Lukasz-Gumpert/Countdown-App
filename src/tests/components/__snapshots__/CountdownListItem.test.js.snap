// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render countdownListItem correctly 1`] = `
<Link
  className="list-item"
  to="edit/undefined"
>
  <p>
     
     
  </p>
  <h1 />
   : 
   
  <p>
    Time left:
  </p>
  <CountdownClock
    timeTillDate={"1970-01-01T00:00:00.000Z"}
  />
</Link>
`;
